asreviewcontrib/semantic_clustering/semantic_clustering.py

```diff
@@ -75,16 +77,50 @@ def SemanticClustering(asreview_data_object):
     print("Running t-SNE...")
     tsne = t_sne(pca, n_iter=1000)
 
+    # calculate optimal number of clusters
+    print("Calculating optimal number of clusters...")
+    n_clusters = calc_optimal_n_clusters(tsne)
+    print("Optimal number of clusters: ", n_clusters)
+
     # run k-means
     print("Running k-means...")
-    labels = run_KMeans(tsne, 10, 10)
+    labels = run_KMeans(tsne, n_clusters, 10)
 
     # visualize clusters
     print("Visualizing clusters...")
     tsne_data = [tsne[:, 0], tsne[:, 1]]
     visualize_clusters(tsne_data, labels)
 
 
+# Optimal n clusters, very inefficient, to be corrected in a future PR
+def calc_optimal_n_clusters(features):
+
+    Sum_of_squared_distances = []
+
+    K = range(1, 25)
```

Comment:
why 25? Should this be a setting that can be adjusted by the user?

Reply:
No, this range is already way over the maximal 'comfortable' amount of clusters, and 25 is only there to limit the algorithm runs. KMeans is quite a light algorithm performance wise and so 25 could be set quite high.